src/hotspot/share/jfr/writers/jfrStreamWriterHost.inline.hpp

```diff
@@ -74,17 +74,13 @@ inline void StreamWriterHost<Adapter, AP>::write_bytes(void* dest, const void* b
 template <typename Adapter, typename AP>
 inline void StreamWriterHost<Adapter, AP>::write_bytes(const u1* buf, intptr_t len) {
   assert(len >= 0, "invariant");
-  while (len > 0) {
-    const unsigned int nBytes = len > INT_MAX ? INT_MAX : (unsigned int)len;
-    const ssize_t num_written = os::write(_fd, buf, nBytes);
-    if (errno == ENOSPC) {
-      JfrJavaSupport::abort("Failed to write to jfr stream because no space left on device", false);
-    }
-    guarantee(num_written > 0, "Nothing got written, or os::write() failed");
-    _stream_pos += num_written;
-    len -= num_written;
-    buf += num_written;
+  const unsigned int nBytes = len > INT_MAX ? INT_MAX : (unsigned int)len;
+  const ssize_t num_written = os::write(_fd, buf, nBytes);
+  if (num_written < 0 && errno == ENOSPC) {
+    JfrJavaSupport::abort("Failed to write to jfr stream because no space left on device", false);
   }
+  guarantee(num_written == 0, "Not all the bytes got written, or os::write() failed");
```

Comment:
guarantee() seems to be a bad way of handling this. I would suggest filing an RFE for more robust error handling.

Reply:
This RFE is created:
https://bugs.openjdk.org/browse/JDK-8307579
The return values of the os::write in jfrStreamWriterHost.inline.hpp:StreamWriterHost<Adapter, AP>::write_bytes(), should be handled more robustly.